### PR TITLE
Support event-mode I(Q)

### LINKED ISSUE #61: Return data in event mode
This is required for polarization but also the type of time-resolved experiments that will run on Zoom.

As this implies normalization in event mode we should probably implement the upper-bound estimation for the variances of the normalized result.

### LINKED ISSUE #64: Loki loader fails to setup variances for events?
```python
def test_sample_event_data_has_variances():
    pipeline = sciline.Pipeline(loki_providers(), params=make_params())
    data = pipeline.compute(sans.types.RawData[sans.types.SampleRun])
    assert data.bins is not None
    assert data.values[0].variances is not None  # AssertionError
```
Please check, but this looks wrong?

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This uses an upper-bound error estimate. Fixes #61.

Also fixes #64.

### CODE REVIEW

File: src/esssans/uncertainty.py [hunk 1]
```diff
@@ -51,3 +51,32 @@ def _no_variance_broadcast(
...
+    da: sc.DataArray, *, events: sc.DataArray
+) -> sc.DataArray:
+    """
+    Upper-bound estimate for errors from normalization in event-mode.
+
+    Count the number of events in each bin of the input data array. Then scale the
+    variances by the number of events in each bin. An explicit broadcast is performed
+    to bypass Scipp's safety check on broadcasting variances.
+
+    Details will be published in an upcoming publication by Simon Heybrock et al.
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/esssans/uncertainty.py:66]: Then we should not use it.

Anyway, I have a very early draft, but it needs more work.

File: src/esssans/__init__.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 )
 from .common import transmission_from_background_run, transmission_from_sample_run
 from .direct_beam import direct_beam
+from .types import BackgroundSubtractedIofQ, IofQ, ReturnEvents, SampleRun
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/esssans/__init__.py:24]: Yes. `types` is meaningless for the user. We should name (and categorize things) in a meaningful way. Yes, there are a lot of things that need updating, I wanted to do that separately, in particular given that there is also other ongoing work that may conflict.

[the remark this answers is not in the document]
Comment (nvaytet) [reply src/esssans/__init__.py:24]: Should we sit down and discuss what would be the best way to do this, so we're all on the same page?

File: src/esssans/uncertainty.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvaytet) [reply src/esssans/uncertainty.py:66]: I was just saying that it is not impossible that you'd find another job before that paper is finished.

File: src/esssans/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/esssans/__init__.py:24]: Definitely!

File: src/esssans/__init__.py [hunk 3]
```diff
@@ -41,6 +42,7 @@
 del importlib
 
 __all__ = [
+    'ReturnEvents',
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/esssans/__init__.py:48]: Fixed!

**Review by nvaytet — APPROVED** [review]
Looks good to me but CI is still failing (probably because of memory usage?)

**Comment by nvaytet** [comment]
> Looks good to me but CI is still failing (probably because of memory usage?)

Does #69 help with fixing the CI I wonder?

**Comment by SimonHeybrock** [comment]
I merged #68, which already fixed the problem, I think this will pass now.

**Pull request merged**